### ISSUE TITLE
[question] Creating `file_browser._picker` with custom setup

### ISSUE THREAD

**Issue by fmorroni** [nvim-telescope/telescope-file-browser.nvim#366]
First of all, sorry for making another issue asking for help. Maybe there should be a discussions thread?

Now to my issue. I want to create a customized file-browser picker to use with another plugin, so that I can replace some of the default actions of the picker (but most should remain as is), and also some options like `path`. The thing is I also want to keep the default picker to use for everyday file browsing.
The default picker I configured with:
```lua
require("telescope").setup({
  extensions = {
    file_browser = {
      theme = "ivy",
      hijack_netrw = true,
      initial_mode = 'normal',
      mappings = M.mappings,
      attach_mappings = function()
        fb_actions.remove:replace(trashy_remove)
        return true
      end,
    },
  },
})
```
For the new picker I tried doing this:
```lua
local zk_opts = {
  initial_mode = 'normal',
  hide_parent_dir = true,
  attach_mappings = function(prompt_bufnr, map)
    -- Just for testing
    actions.select_default:replace(function()
      local selection = action_state.get_selected_entry()
      P(selection[1])
    end)
    return true
  end,
  mappings = mappings,
  path = vim.env.ZK_NOTEBOOK_DIR,
}

local zk_picker = function()
  require('telescope').extensions.file_browser._picker(zk_opts)
end

return zk_picker
```
And this actually mostly works, the only thing that doesn't seem to work are the custom key mappings.
The `mappings` object I'm passing is the same as for the original's setup, and it works there, so not sure what could be the problem...

**Comment by fmorroni** [comment]
Found another problem. When using this custom picker, when I go into a directory and the picker refreshes to update the entries, the custom picker closes and the default file-browser picker opens.

https://github.com/nvim-telescope/telescope-file-browser.nvim/assets/88512894/51c6bd3e-71b8-48e1-90a5-edc269da0e98

The first picker I opened was my custom one. After I select `dir1`, the custom picker closes and the default one opens.

**Comment by jamestrew** [comment]
For attaching mappings to picker, we currently only allow this via the `attach_mappings` pattern. This is a core telescope thing. The `mappings` option is really only for the top level setup function.

For replacing `select_default`, I think you just have to use `require('telescope').extensions.file_browser.file_browser` rather than the "private" `_picker.

Let me know if you're still having issue.

**Comment by fmorroni** [comment]
Okay you're totally right on both things.

I had read about the `attach_mappings` on the [developers documentation](https://github.com/nvim-telescope/telescope.nvim/blob/master/developers.md) and they didn't really explain how the `map` parameter worked so I wasn't sure how to use it. Now I checked at `:h telescope.map` and it was explained there... my bad.

About using `require('telescope').extensions.file_browser.file_browser` instead of `_picker`, I had actually tried it but the default picker opened. I didn't realize I could pass a new configuration as an argument.
And I was about to say that changing the theme didn't work, because doing `require('telescope').extensions.file_browser.file_browser({theme = 'dropdown'})` still opened it as `ivy` as the default picker. But apparently the `theme` option is like the `mappings` one and only works on the top level setup.
Using `file_browser(require('telescope.themes').get_dropdown(zk_opts))` worked.

Thanks for the help again!
